### PR TITLE
OWDatasets: PyLint

### PR DESCRIPTION
**Pull request by janezd**
Now passes pylint with disabling only `missing-docstring`, `no-name-in-module`, `too-many-ancestors`, `no-else-return`.

### CODE REVIEW

File: Orange/widgets/data/owdatasets.py [hunk 1]
```diff
@@ -522,15 +528,17 @@ def onDeleteWidget(self):
...
         return QSize(900, 600)
 
     def closeEvent(self, event):
         self.splitter_state = bytes(self.splitter.saveState())
         self.header_state = bytes(self.view.header().saveState())
         super().closeEvent(event)
 
-    def load_data(self, path):
+    @staticmethod
+    def load_data(path):
```
Comment (lanzagar) [inline Orange/widgets/data/owdatasets.py]: This method does not do much here, but is overridden (at least) in orange3-single-cell. It is not a staticmethod there, so we might want to keep it non-static here as well for consistency.

File: Orange/widgets/data/owdatasets.py [hunk 2]
```diff
@@ -171,11 +172,17 @@ class Outputs:
...
         self.local_cache_path = os.path.join(data_dir(), self.DATASET_DIR)
 
-        self._header_labels = [header['label'] for _, header in self.HEADER_SCHEMA]
-        self._header_index = namedtuple('_header_index', [info_tag for info_tag, _ in self.HEADER_SCHEMA])
-        self.Header = self._header_index(*[index for index, _ in enumerate(self._header_labels)])
+        self._header_labels = [
+            header['label'] for _, header in self.HEADER_SCHEMA]
+        self._header_index = namedtuple(
+            '_header_index', [info_tag for info_tag, _ in self.HEADER_SCHEMA])
+        self.header = self._header_index(
```
Comment (lanzagar) [inline Orange/widgets/data/owdatasets.py]: Changing `Header` to `header` breaks the scOrange datasets widget, which extends this one and redefines `Header`.

I see no reason why it was capitalized before, so I agree with the change. But it needs to be coordinated with changes in widgets that extend this one.

File: Orange/widgets/data/owdatasets.py (hunk 1 above)
Comment (janezd) [reply Orange/widgets/data/owdatasets.py]: Agreed.

**Comment by janezd** [comment]
I didn't know that this widget is extended. But that's the reason why I only self-merged the PRs with trivial changes.

I would prefer if we decide to never subclass widget classes. Better still, we should never import a module with a widget, unless this is an abstract widget class. Let's discuss this on Friday, and if we agree, I'll make widget classes final unless explicitly declared as open (with a reasonable period of deprecation warnings). NB, all Kotlin classes are final by default.

Argument: if widgets can be subclassed, our hands are rather tied when it comes to any redesigning. Touch any widget and derived widgets will break.

**Pull request merged**